Dockerfile

```diff
@@ -1,11 +1,17 @@
-FROM node:latest
+FROM node:alpine
+
+ARG VERSION
+ARG TYPE=basic
+
+ENV NODE_ENV production
 
 RUN mkdir -p /usr/src/app
 WORKDIR /usr/src/app
 
-COPY package.json /usr/src/app/
-RUN npm install
-COPY . /usr/src/app
+RUN npm install express-gateway@$VERSION && \
+    ./node_modules/.bin/eg gateway create -n gateway -d . -t $TYPE && \
+    npm cache clean --force
```

Comment:
@rlisagor Why the cache clean?

Reply:
This is to reduce the size of the resulting image as much as possible.